cql3/untyped_result_set.cc

```diff
@@ -38,7 +52,7 @@ bool cql3::untyped_result_set_row::has(std::string_view name) const {
 cql3::untyped_result_set_row::view_type cql3::untyped_result_set_row::get_view(std::string_view name) const {
     auto& data = _data.at(index(name));
     if (!data) {
-        throw std::bad_variant_access();
+        throw missing_column(name);
```

Comment:
this might not be very accurate either. in #14944, the result set contains no rows at all. not to mention the column.

Reply:
This exception is thrown from a method of `untyped_result_set_row`. So, a row has to exist and I think the name of the exception makes sense.

As to why it was thrown in this particular scenario... I'm not sure. If the result set were really empty then I think a different exception would be thrown, from other methods that are called before `get_as`. The result set is generated from a query that selects a static column, so maybe a dummy row is created. I'll check.

Reply:
just to be clear, the query i mentioned was

```
"SELECT count(range_end) as cnt, num_ranges FROM system.{} WHERE id = ?",
                           CDC_GENERATIONS_V3),
```
if i understand correctly, `cnt` should always exist, but accordingly to my test `num_ranges` did not.

Reply:
Ok, the reason is a bit different. The query is an aggregation over the clustering rows:

```
SELECT count(range_end) as cnt, num_ranges FROM system.{} WHERE id = ?", CDC_GENERATIONS_V3)
```

I did an experiment and ran a similar query on an empty table:

```
cqlsh> create table ks.t (pk int, ck int, c int, s int static, primary key(pk, ck));
cqlsh> select count(c) as cnt, s from ks.t where pk = 0;

 cnt | s
-----+------
   0 | null

(1 rows)
```

So, the result will contain one row even if the queried partition is empty.

Reply:
ahh, thank you for the explanation. so we consider the aggregated result as a row even if some of the fields are not available.

now, `missing_column` does make sense to me now! 

sorry for the noise.